homeassistant/components/ping/binary_sensor.py

```diff
@@ -57,13 +55,32 @@ async def async_setup_platform(
     async_add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
-    """Set up the Ping Binary sensor."""
+    """Init Ping sensor from YAML."""
+
+    setup_sensor(hass, config, async_add_entities)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up a Ping config entry."""
+
+    setup_sensor(hass, entry.options, async_add_entities, entry, entry.title)
+
+
+def setup_sensor(
```

Comment:
```suggestion
@callback
def async_setup_sensor(
```

Since we have both sync and async code in this code base, please prefix function that are safe to run in the event loop with `async_`